astropy/time/core.py

```diff
@@ -803,12 +803,15 @@ def _apply(self, method=None, *args, **kwargs):
             # Apply the method to any value arrays (though skip if there is only
             # a single element and the method would return a view, since in
             # that case nothing would change).
-            if method is not None and val is not None:
-                if method == 'copy' or method == 'flatten' and val.size == 1:
+            if method is not None and val is not None and hasattr(val, method):
+                if val.size > 1 or method == 'copy':
```

Comment:
The implicit requirement here is that if `val` has the specified `method`, then it is actually a numpy array with a `size` property.  Still, maybe it would be worth using `if getattr(val, 'size', 0) > 1 ...`?